samples/dnn/mask_rcnn.cpp

```diff
@@ -0,0 +1,220 @@
+        double freq = getTickFrequency() / 1000;
+        double t = net.getPerfProfile(layersTimes) / freq;
+        string label = format("openCV Mask R-CNN : %0.0f ms", t);
+        putText(frame, label, Point(0, 15), FONT_HERSHEY_SIMPLEX, 0.5, Scalar(0, 0, 0));
+
+        Mat detectedFrame;
+        frame.convertTo(detectedFrame, CV_8U);
+        imshow(kWinName, frame);
+
+    }
+
+    cap.release();
+    if (!parser.has("image")) video.release();
+
+    return 0;
+}
+
+void postprocess(Mat& frame, const vector<Mat>& outs)
+{
+    Mat outDetections = outs[0];
+    Mat outMasks = outs[1];
+
+    // Output size of masks is NxCxHxW where
+    // N - number of detected boxes
+    // C - number of classes (excluding background)
+    // HxW - segmentation shape
+    const int numDetections = outDetections.size[2];
+
+    outDetections = outDetections.reshape(1,(int)(outDetections.total() / 7));
+    for ( int i = 0; i < numDetections; ++i)
+    {
+        float score = outDetections.at<float>(i, 2);
+        if (score > confThreshold)
+        {
+            int classId = static_cast<int>(outDetections.at<float>(i, 1));
+            int left = static_cast<int>(frame.cols * outDetections.at<float>(i, 3));
+            int top = static_cast<int>(frame.rows * outDetections.at<float>(i, 4));
+            int right = static_cast<int>(frame.cols * outDetections.at<float>(i, 5));
+            int bottom = static_cast<int>(frame.rows * outDetections.at<float>(i, 6));
+
+            left = max(0, min(left, frame.cols - 1));
+            top = max(0, min(top, frame.rows - 1));
+            right = max(0, min(right, frame.cols - 1));
+            bottom = max(0, min(bottom, frame.rows - 1));
+            Rect box = Rect(left, top, right - left + 1, bottom - top + 1);
+
+            Mat objectMask(outMasks.size[2], outMasks.size[3],CV_32F, outMasks.ptr<float>(i,classId));
+
+            drawBox(frame, classId, score, box, objectMask);
+        }
+    }
+}
+
+void drawBox(Mat& frame, int classId, float conf, Rect box, Mat& objectMask)
+{
+    rectangle(frame, Point(box.x, box.y), Point(box.x+box.width, box.y+box.height), Scalar(255, 178, 50), 3);
+
+    string label = format("%.2f", conf);
+    if (!classes.empty())
+    {
+        CV_Assert(classId < (int)classes.size());
+        label = classes[classId] + ":" + label;
+    }
+
+    int baseLine;
+    Size labelSize = getTextSize(label, FONT_HERSHEY_SIMPLEX, 0.5, 1, &baseLine);
+    box.y = max(box.y, labelSize.height);
+    rectangle(frame, Point(box.x, box.y - (int)round(1.5*labelSize.height)), Point(box.x + (int)round(1.5*labelSize.width), box.y + baseLine), Scalar(255, 255, 255), FILLED);
+    putText(frame, label, Point(box.x, box.y), FONT_HERSHEY_SIMPLEX, 0.75, Scalar(0,0,0),1);
+
+    Scalar color = colors[classId%colors.size()];
+
+    resize(objectMask, objectMask, Size(box.width, box.height));
+    Mat mask = (objectMask > maskThreshold);
+    Mat coloredRoi = (0.3 * color + 0.7 * frame(box));
```

Comment:
Suggest to get rid of extra buffers to simplify th code:
```
cv::Mat roi = frame(box);
roi = roi * 0.7 + color * 0.3;
drawContours(roi, ...);
```

Reply:
done